examples/contrib/readhar.py

```diff
@@ -0,0 +1,141 @@
+"""Reads HAR files into flow objects"""
+import asyncio
+import json
+import logging
+import time
+
+from mitmproxy import command
+from mitmproxy import connection
+from mitmproxy import ctx
+from mitmproxy import exceptions
+from mitmproxy import http
+from mitmproxy import types
+
+logger = logging.getLogger(__name__)
+
+
+class ReadHar:
+    def __init__(self):
+        self.flows = []
+
+    def fix_headers(
+        self, request_headers: list[dict[str, str]] | list[tuple[str, str]]
+    ) -> list[tuple[bytes, bytes]]:
+        """Converts provided headers into (b"header-name", b"header-value") tuples"""
+        flow_headers = []
+        for header in request_headers:
+            # Applications that use the {"name":item,"value":item} notation are Brave, Chrome, Edge, Firefox, Charles, Fiddler, Insomnia, Safari
+            if isinstance(header, dict):
+                try:
+                    key = header["name"]
+                    value = header["value"]
+                except KeyError as e:
+                    raise exceptions.OptionsError(str(e)) from e
+
+            # Application that use the [name, value] notation is Slack
+            else:
+                try:
+                    key = header[0]
+                    value = header[1]
+                except IndexError as e:
+                    raise exceptions.OptionsError(str(e)) from e
+            flow_headers.append((bytes(key, "utf-8"), bytes(value, "utf-8")))
+
+        return flow_headers
+
+    # Don't know how to make a type annotation for the request json
+    def request_to_flow(self, request_json: dict) -> http.HTTPFlow:
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        request_method = request_json["request"]["method"]
+        request_url = request_json["request"].get("url", None)
+        if not request_url:
+            request_url = request_json["request"].get("host", None)
```

Comment:
Can you explain this part here? When is `url` unavailable? And does `host` really contain a URL in this case?